Mod1/Lessons/Week5/Scope.md

```diff
@@ -0,0 +1,232 @@
+# Scope and Access Modifiers
+
+## Learning Goals
+* Identify how **scope** affects where variables can be accessed
+* Have a reference to look at when **scope** errors show up
+* Understand the difference between the **public** and **private** access modifiers
```

Comment:
I realized while working on this that when to use public vs private is actually quite a nuanced and advanced topic. Because it works to leave everything public, it's just better to make things that should be private, private. 

I think it's good to start thinking about how others interact with your code, not sure if by week 5 students will be ready? I'm thinking I'll split this into two lessons and we can drop them into the schedule when we think students are ready for them?

Reply:
I agree - this can be very nuanced.  To me, there are two 'learning goals' around public/private:
  * definition: relatively simple, Mod1 students should be ready to master this.
  * use cases: very nuanced, something that could be another lesson in a later mod, or something that students build up through practice and on the job.

Reply:
Ooo I like this distinction. I agree, we want everyone to understand the different and start thinking about the uses cases but it almost certainly will take a while to make sense.